### DISCUSSION TITLE
Issue when setSheetContent - Vertex for address missing in AddressMapping/ Cannot read properties of undefined (reading 'size')

### DISCUSSION THREAD

**Discussion by alvarovillafane** [handsontable/hyperformula#1229] [API Questions]
Hi, 

I'm getting `Vertex for address missing in AddressMapping` when using a `setSheetContent` and I'd like to understand why is happening. 

Here is the codesandbox:
https://codesandbox.io/s/mystifying-ardinghelli-o136jr?file=/src/index.js:0-756

This is the code that is triggering the issue:
```
import HyperFormula from "hyperformula";

///Vertex for address missing in AddressMapping

const newSheet = ["=A1+B5"];
const newRow = ["=A1+B5"];
const hfInstance = HyperFormula.buildFromArray([newSheet], {
  licenseKey: "gpl-v3"
});

const sheetId = 0;
hfInstance.addRows(sheetId, [1, 1]);
hfInstance.setCellContents({ sheet: sheetId, col: 0, row: 1 }, newRow[0]);
hfInstance.setSheetContent(sheetId, [newSheet, newRow]);

```

And also getting this issue `Cannot read properties of undefined (reading 'size')`

With a slightly different code.

```
// Cannot read properties of undefined (reading 'size')
const newSheet = ['=A1+B5'];
const newRow = ['=A1+B5'];
const hfInstance = HyperFormula.buildFromArray([newSheet], { licenseKey: 'gpl-v3' });

const sheetId = 0;
hfInstance.addRows(sheetId, [1, 1]);
hfInstance.setSheetContent(sheetId, [newSheet, newRow]);
```

Can someone shine a light and give some insight into what's happening?

Thanks for the help!

**Comment by AMBudnik** [comment]
Thank you for sharing the issue report along with the code snippet, @vmalvaro 
I replicated the same error. 

While debugging the demo, I found out that
* replacing `newSheet` value with a non-reference content (ex. `=45`), or settings up any possible address for the 1st column like `=C1` or `=GX1` prevents from getting the error
* commenting out `addRows()` method also prevents the error from showing

**Reply by phassis99** [reply]
@AMBudnik  when using the 'addRows()' function, it allows for updating the table cell references to shift one row down, even if the original references were invalid. This behavior is similar to what occurs in a spreadsheet when you drag down an invalid reference.

For example: 
```.ts
const newSheet = ['=A1+B5'];
const newRow = ['=A1+2'];
const hfInstance = HyperFormula.buildFromArray([newSheet], { licenseKey: 'gpl-v3' });
const sheetId = 0;
hfInstance.addRows(sheetId, [1, 1]);
hfInstance.setCellContents({ sheet: sheetId, col: 0, row: 1 }, newRow[0]);

console.log(hfInstance.getAllSheetsSerialized()); 
```
the console above should return = { Sheet1: [ [ '=A1+B6' ], [ '=A1+2' ] ] };
Which is what we expect.
But then later, when we try to use `hfInstance.setSheetContent(sheetId, [newSheet, newRow])` , we get the vertex error.

Essentially, what we are trying to do is:
- Users should be able to create a formula even if it involves a row that doesn't exist, like A1+B5.
- If a user adds a new row, the cell references in the formula should update accordingly, even if it was initially an invalid reference. For example, after inserting a row below, the formula should change to A1+B6.


Please let us know if the explanation provided makes sense to you. We appreciate your time and help with this issue🙏.

**Reply by AMBudnik** [reply]
It is fine, @vmalvaro You should be able to run both of those methods without getting errors.
Our developer did a quit check of this issue and it might be related to https://github.com/handsontable/hyperformula/issues/1188

**Comment by sequba** [comment]
@vmalvaro @imphassis This issue will be fixed in the upcoming release as it has the same cause as #1188

**Comment by sequba** [comment]
@vmalvaro @imphassis the fix was just released with HyperFormula 2.4.0

**Reply by phassis99** [reply]
Thanks for it @sequba!
